paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonScanBuilder.scala

```diff
@@ -33,4 +40,45 @@ class PaimonScanBuilder(table: Table)
     // just make a best effort to push down limit
     false
   }
+
+  override def supportCompletePushDown(aggregation: Aggregation): Boolean = {
+    // for now we only support complete push down, so there is no difference with `pushAggregation`
+    pushAggregation(aggregation)
+  }
+
+  // Spark does not support push down aggregation for streaming scan.
+  override def pushAggregation(aggregation: Aggregation): Boolean = {
+    if (localScan.isDefined) {
+      return true
+    }
+
+    // Only support with push down partition filter
+    if (pushed.length != partitionFilter.length) {
```

Comment:
Can we also save `postScanFilters` in `pushFilters`, and just check is empty here.

Reply:
+1, added `postScanFilters `